### PR TITLE
Schwork

### PR DESCRIPTION
**Pull request by modelB**
resolvers merged w modularized schema

### CODE REVIEW

File: server/controllers/gqlController.js [hunk 1]
```diff
@@ -108,21 +109,118 @@ gqlController.makeSchemaQueries = async function (req, res, next) {
...
+        }
+        output += "\n}";  
+        res.locals.resolvers = output;
+        return next();
+        // return resolvers;  
+    } catch (err) {
+        console.log('ERR IN MAKERESOLVERS');
+        return next(err);
+    }
+}
```
Comment (AdamrG1) [inline server/controllers/gqlController.js:223]: Thoughts on modularizing this function and importing into the controller? Similar to the controllers above?

File: server/controllers/gqlController.js [hunk 2]
```diff
@@ -108,21 +109,118 @@ gqlController.makeSchemaQueries = async function (req, res, next) {
...
+        const resolvers = {};
+        resolvers['Query'] = {};
+        resolvers['Mutation'] = {};
+        const nonJoinTables = {};
+        const joinTables = {};
+        //loop thru all table data
+        for (let [key,val] of Object.entries(res.locals.data.allTables)){
+          //count props in fks that match current key
+              const fkCount = res.locals.data.foreignKeys.filter(el => el.foreign_table === key).length;
+              //if not a join table
```
Comment (AdamrG1) [inline server/controllers/gqlController.js:123]: Possible to import func for nonJoinTables

File: server/controllers/gqlController.js [hunk 3]
```diff
@@ -108,21 +109,118 @@ gqlController.makeSchemaQueries = async function (req, res, next) {
...
+            if (val.length > fkCount+1) {
+              nonJoinTables[key] = true;
+              //query one
+              //look for pk name
+              let pkName;
+              for (let i = 0; i<res.locals.data.primaryKeys.length; i++) {
+                  if (res.locals.data.primaryKeys[i].table_name === key) pkName = res.locals.data.primaryKeys[i].primary_key_columns;
+              }
+              let queryKey = singular(key) === key ? `${camelCase(key)}ById` : camelCase(key);
```
Comment (AdamrG1) [inline server/controllers/gqlController.js:132]: Add comments to describe the purpose of the line i.e. what/why/how

File: server/controllers/gqlController.js [hunk 4]
```diff
@@ -108,21 +109,118 @@ gqlController.makeSchemaQueries = async function (req, res, next) {
...
+                joinTables[key] = true;
+            }
+        }
+        //loop through pks and add as values in joinTables and nonJoinTables for ref
+        for (let k = 0; k< res.locals.data.primaryKeys.length; k++) {
+            const pkObjRef = res.locals.data.primaryKeys[k];
+            if (pkObjRef.table_name in nonJoinTables) nonJoinTables[pkObjRef.table_name] = pkObjRef.primary_key_columns;
+            else joinTables[pkObjRef.table_name] = pkObjRef.primary_key_columns;
+        }
+        //loop through nonjointables
```
Comment (AdamrG1) [inline server/controllers/gqlController.js:167]: add comments for what the purpose of this code is

File: server/controllers/gqlController.js [hunk 5]
```diff
@@ -108,21 +109,118 @@ gqlController.makeSchemaQueries = async function (req, res, next) {
...
+              output += (`\n    ${innerKey}: `+ innerVal);
+            }
+            output += `\n  },`;
+        }
+        output += "\n}";  
+        res.locals.resolvers = output;
+        return next();
+        // return resolvers;  
+    } catch (err) {
+        console.log('ERR IN MAKERESOLVERS');
```
Comment (AdamrG1) [inline server/controllers/gqlController.js:220]: add the err message as well, (`Err in makeResolvers`, err)

**Review by AdamrG1 — DISMISSED** [review]
Code looks good to me. Can you add comments to the code explaining what/why. Also add the error message in the catch block.

**Review by markyliu — APPROVED** [review]
lgtm, next steps: comments later, convert front to using data built on back

**Review by Moonwalker5823 — APPROVED** [review]
LGTM

**Pull request merged**